Fix for bug where an array as generic type argument will cause an exception in GetSchema

ASuurkuusk opened the pull request:
If TypeModel.GetSchema is called on a TypeModel where a generic type has an array as type argument, then an exception is thrown. 

This PR includes two commits, one that includes tests that show the bug, and one includes a suggested fix for the bug.

mgravell commented:
Have you tested how this works with things like `byte[]`? I'm thinking of scenarios where it isn't the generic argument; possible to add a test to show that this is unaffected? ditto when arrays are interpreted as `repeated`?

mgravell commented:
(clarification: I'm not saying I think it will misbehave; I'm saying that *because it is even a question on my mind*, it probably warrants a test)

ASuurkuusk commented:
I added a few more tests to verify that the schema is correctly generated for byte[] (bytes), int[] (repeated Int32), and SimpleClass[] (repeated SimpleClass), both for generic type fields and ordinary fields.

I also added a test to verify that NotSupportedException is correctly thrown for multi-dimensional and jagged arrays. However this test fails for byte[,] and byte[][], but this does not seem related to my fix. It's possible to generate schemas for byte[,] and byte[][] fields in a "normal" type and the schema will contain a "bytes" member. However, an exception will be thrown when trying to serialize such a type.

mgravell commented:
Seems legit; merging - thanks
